designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -327,12 +328,20 @@ class DeploymentServiceImpl(
     } yield result)
   }
 
+  override def getProcessesInProgress: Future[Map[ProcessId, Set[ProcessActionType]]] =
+    dbioRunner.run(
+      actionRepository.getInProgressCanceledAndDeployed
+    )
+
   override def getProcessState(
-      processDetails: BaseProcessDetails[_]
+      processDetails: BaseProcessDetails[_],
+      processesInProgress: Option[Map[ProcessId, Set[ProcessActionType]]] = None
   )(implicit user: LoggedUser, ec: ExecutionContext, freshnessPolicy: DataFreshnessPolicy): Future[ProcessState] = {
     dbioRunner.run(for {
-      inProgressActionTypes <- actionRepository.getInProgressActionTypes(processDetails.processId)
-      result                <- getProcessState(processDetails, inProgressActionTypes)
+      inProgressActionTypes <- processesInProgress.fold(
```

Comment:
I hate this concept of folding options and booleans. WDYT about replacing it by `map().getOrElse()`?

Reply:
not applicable any more